src/main/java/Duke.java

```diff
@@ -1,10 +1,64 @@
+import Exceptions.DukeException;
+import java.util.Scanner;
 public class Duke {
+    private final Ui _ui;
+    private TaskList _taskList;
+    private final Storage _storage;
+    public static final String DATA_FILE_PATH = "data/duke.txt";
+
+    public Duke(String filepath) {
+        _ui = new Ui();
+        _storage = new Storage(filepath);
+        try {
+            _taskList = new TaskList(_storage.load());
+        } catch (DukeException e) {
+            _ui.showDukeError(e);
+            _taskList = new TaskList();
+        }
+    }
+
+    public void run() {
+        _ui.showWelcome();
+        // Create a scanner object to read commands entered by the user
+        Scanner scanner = new Scanner(System.in);
+
+        // Start command loop
+        while (true) {
+            // Read the next line of input
+            String userInput = scanner.nextLine();
+
+            try {
+                if (userInput.equals("bye")) {
```

Comment:
Perhaps try refactoring this into a switch statement which may result in a cleaner-looking code. Good to standardise whether you want to use if-else or switch statements throughout your code!